cctp_primitives/src/proving_system/verifier/certificate.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{
+    type_mapping::{FIELD_SIZE, FieldElement},
+    proving_system::{
+        verifier::{UserInputs, ZendooVerifier},
+        error::ProvingSystemError,
+    },
+};
+use crate::utils::get_cert_data_hash;
+
+/// All the data needed to reconstruct the aggregated input for the NaiveThresholdSignatureCircuit
+/// included in the Certificate.
+pub struct CertificateProofUserInputs<'a> {
+    constant:                                   Option<&'a [u8; FIELD_SIZE]>,
+    epoch_number:                               u32,
+    quality:                                    u64,
+    bt_list:                                    &'a [(u64,[u8; 20])],
+    custom_fields:                              Option<&'a [[u8; FIELD_SIZE]]>,
+    end_cumulative_sc_tx_commitment_tree_root:  &'a [u8; FIELD_SIZE],
+    btr_fee:                                    u64,
+    ft_min_fee:                                 u64
+}
+
+impl UserInputs for CertificateProofUserInputs<'_> {
+    fn get_circuit_inputs(&self) -> Result<Vec<FieldElement>, ProvingSystemError> {
+
+        let aggregated_input = get_cert_data_hash(
+            self.constant, self.epoch_number, self.quality, self.bt_list, self.custom_fields,
+            self.end_cumulative_sc_tx_commitment_tree_root, self.btr_fee, self.ft_min_fee
+        ).map_err(|e| ProvingSystemError::Other(format!("{:?}", e)))?;
+
+        Ok(vec![aggregated_input])
+    }
+}
+
+pub struct ZendooCertProofVerifier<'a>(std::marker::PhantomData<&'a ()>);
```

Comment:
It would be useful having also a test section which uses this verifier, even with a stubbed implementation, analogously to what has been done in random_batch_verifier_test() with the TestCircuitVerifier obj.